pred.py

```diff
@@ -2,98 +2,111 @@
 from datasets import load_dataset
 import torch
 import json
-from transformers import AutoTokenizer, LlamaTokenizer, LlamaForCausalLM, AutoModelForCausalLM
+from transformers import AutoConfig, AutoTokenizer, LlamaTokenizer, LlamaForCausalLM, AutoModelForCausalLM
 from tqdm import tqdm
 import numpy as np
 import random
 import argparse
 from llama_flash_attn_monkey_patch import replace_llama_attn_with_flash_attn
 import torch.distributed as dist
 import torch.multiprocessing as mp
+from transformers import (DynamicCache,
+                          SinkCache,
+                          SlidingWindowCache,
+                          QuantoQuantizedCache,
+                          )
+import gc
+from vllm import LLM, SamplingParams
 
 def parse_args(args=None):
     parser = argparse.ArgumentParser()
-    parser.add_argument('--model', type=str, default=None, choices=["llama2-7b-chat-4k", "longchat-v1.5-7b-32k", "xgen-7b-8k", "internlm-7b-8k", "chatglm2-6b", "chatglm2-6b-32k", "chatglm3-6b-32k", "vicuna-v1.5-7b-16k", "llama3.1-8b-instruct", "baselineWithCuratedFTLongMSL8K", "baselineWithCuratedFTLongMSL8KCQ"])
+    parser.add_argument('--model', type=str, default=None,)
+    parser.add_argument('--max_seq_len', type=int, default=7500,)
     parser.add_argument('--e', action='store_true', help="Evaluate on LongBench-E")
+    parser.add_argument('--kv_method', type=str, default='full',)
+    parser.add_argument('--prefill_kv', action='store_true', help="prefill kv cache before generate function")
+    parser.add_argument('--backend', type=str, default='hf',choices=["hf","vllm"])
+    parser.add_argument('--load_in_8bit', action='store_true')
     return parser.parse_args(args)
 
-# This is the customized building prompt for chat models
-def build_chat(tokenizer, prompt, model_name):
-    if "chatglm3" in model_name:
-        prompt = tokenizer.build_chat_input(prompt)
-    elif "chatglm" in model_name:
-        prompt = tokenizer.build_prompt(prompt)
-    elif "longchat" in model_name or "vicuna" in model_name:
-        from fastchat.model import get_conversation_template
-        conv = get_conversation_template("vicuna")
-        conv.append_message(conv.roles[0], prompt)
-        conv.append_message(conv.roles[1], None)
-        prompt = conv.get_prompt()
-    elif "llama2" in model_name:
-        prompt = f"[INST]{prompt}[/INST]"
-    elif "xgen" in model_name:
-        header = (
-            "A chat between a curious human and an artificial intelligence assistant. "
-            "The assistant gives helpful, detailed, and polite answers to the human's questions.\n\n"
-        )
-        prompt = header + f" ### Human: {prompt}\n###"
-    elif "internlm" in model_name:
-        prompt = f"<|User|>:{prompt}<eoh>\n<|Bot|>:"
-    return prompt
+def prefill_kv_cache(past_key_values,model,input,max_gen_len):
```

Comment:
nit: formatting (space after comma - maybe worth running black in a separate PR?)
Also, despite the original code is not well documented, I'd add docstrings with brief description.